Sources/ArcGISToolkit/Components/OfflineMapAreasView/PreplannedMapModel.swift

```diff
@@ -17,15 +17,60 @@ import ArcGIS
 
 /// An object that encapsulates state about a preplanned map.
 @MainActor
-class PreplannedMapModel: ObservableObject, Identifiable {
+public class PreplannedMapModel: ObservableObject, Identifiable {
     /// The preplanned map area.
     let preplannedMapArea: any PreplannedMapAreaProtocol
     
+    /// The task to use to take the area offline.
+    private let offlineMapTask: OfflineMapTask?
+    
+    /// The download directory for the preplanned map areas.
+    private let preplannedDirectory: URL?
+    
+    /// The ID of the preplanned map area.
+    private var preplannedMapAreaID: String?
+    
+    /// The mobile map package for the preplanned map area.
+    private(set) var mobileMapPackage: MobileMapPackage?
+    
+    /// The currently running download job.
+    @Published private(set) var job: DownloadPreplannedOfflineMapJob?
+    
     /// The combined status of the preplanned map area.
     @Published private(set) var status: Status = .notLoaded
     
-    init(preplannedMapArea: PreplannedMapAreaProtocol) {
+    /// The result of the download job. When the result is `.success` the mobile map package is returned.
+    /// If the result is `.failure` then the error is returned. The result will be `nil` when the preplanned
+    /// map area is still packaging or loading.
+    @Published private(set) var result: Result<MobileMapPackage, Error>?
+    
+    /// A Boolean value indicating if download can be called.
+    var canDownload: Bool {
+        switch status {
+        case .notLoaded, .loading, .loadFailure, .packaging, .packageFailure,
+                .downloading, .downloaded:
+            false
+        case .packaged, .downloadFailure:
+            true
+        }
+    }
+    
+    init(
+        preplannedMapArea: PreplannedMapAreaProtocol,
```

Comment:
preplanned might be repeating type information. I'd probably just call the parameter `mapArea`.